### DISCUSSION TITLE
How can I avoid component unmount if route changes with v6?

### DISCUSSION THREAD

**Discussion by jaybowang** [remix-run/react-router#9269] [Q&A]
![image](https://user-images.githubusercontent.com/1948533/190419434-4ebf8034-56c0-46e9-9900-c49372073c77.png)

How to keep the state when a user switches between these menus?

I found [a demo with v5](https://codesandbox.io/s/persist-state-react-router-1t4bl) from [stackoverflow](https://stackoverflow.com/a/45930922/1573750). 

Haven't figured out how to do this with v6. Googled and tried the examples but still have no idea.

**Comment by luckydevboy** [comment]
According to the [upgrading from v5](https://reactrouter.com/en/main/upgrading/v5#prompt-is-not-currently-supported) `Prompt` and `useBlocker` are no longer supported.

**Comment by userAugustos** [comment]
@jaybowang find any solution for this?
